### PR TITLE
Small usability improvements on the TileMap editor tile palette.

### PR DESCRIPTION
**Pull request by ghost**
- The column width changes with the zoom, now the icons don't overlap.
- Added a tile hsepation setting.
- Added a setting to show/hide tile names.

| Before | Now |
| --- | --- |
| tile_map_palette_before | tile_map_palette_after |

### CODE REVIEW

**Comment by bojidar-bg** [comment]
Looks pretty good

File: tools/editor/plugins/tile_map_editor_plugin.cpp [hunk 1]
```diff
@@ -205,16 +212,16 @@ void TileMapEditor::_update_palette() {
 	if (tiles.empty())
 		return;
 
+	float min_size = EDITOR_DEF("tile_map/preview_size", 64);
+	min_size *= EDSCALE;
+	int hseparation = EDITOR_DEF("tile_map/palette_item_hseparion",8);
```
Comment (akien-mga) [inline tools/editor/plugins/tile_map_editor_plugin.cpp]: Typo: palette_item_hsepar**at**ion.

File: tools/editor/plugins/tile_map_editor_plugin.cpp [hunk 2]
```diff
@@ -1458,6 +1472,9 @@ void TileMapEditorPlugin::make_visible(bool p_visible) {
 TileMapEditorPlugin::TileMapEditorPlugin(EditorNode *p_node) {
 
 	EDITOR_DEF("tile_map/preview_size",64);
+	EDITOR_DEF("tile_map/palette_item_hseparion",8);
```
Comment (akien-mga) [inline tools/editor/plugins/tile_map_editor_plugin.cpp]: Same typo as above.

File: tools/editor/plugins/tile_map_editor_plugin.cpp [hunk 3]
```diff
@@ -228,10 +235,14 @@ void TileMapEditor::_update_palette() {
 			name = "#"+itos(E->get());
 		}
 
-		if (filter != "" && name.findn(filter) == -1)
+		if (filter != "" && !filter.is_subsequence_ofi(name))
```
Comment (akien-mga) [inline tools/editor/plugins/tile_map_editor_plugin.cpp]: A fourth unmentioned usability improvement ;)

**Comment by akien-mga** [comment]
Good to merge once the typo is fixed.

**Comment by TheHX** [comment]
I fixed the typo

**Comment by bojidar-bg** [comment]
YAY

**Pull request merged**